text/0059-token-metadata-document.md

```diff
@@ -0,0 +1,275 @@
+* `token_id`: A decimal string representation of the `u64` corresponding to the token id.
+* `currency_name`: A UTF-8 string which is the full, official name of the currency. This is intended to be an English language noun or noun phrase.
+* `short_code`: A short code for the currency, up to 12 ASCII characters matching `/[A-Za-z0-9.-]{1,12}/`. This may be displayed to users in a manner similar to ISO 4217 currency codes, for example, USD, GBP, CAD, in connection to an amount of currency. This is expected to match the ticker symbol used for this asset on cryptocurrency exchanges.
+* `symbol`: Optional UTF-8 string. Most fiat currencies have a printable character such as $, £, ¥. Some cryptocurrencies do also. Bitcoin has ₿. Ethereum has Ξ.
+* `decimals`: An integer specifying how the `u64` integer in a TxOut in the blockchain is scaled to compute a user-displayed amount of the currency. For example, MOB has 12 decimals, which indicates that `10^12` of the smallest representable units of MOB on the MobileCoin blockchain are equal to one MOB.
+* `logo_svg`: An optional logo image. This is a base64-encoded SVG document. This is expected to have been sanitized using something like [svg-hush](https://github.com/cloudflare/svg-hush), to remove scripting, hyperlinks to other documents, and references to cross-origin resources. The full extent of such sanitization will not be specified here.
+* `info_url`: A link to a website containing more information about the token that may be interesting to token holders. This should have basic information about the purpose of the token, its supply, any utility that it has, or links to associated whitepapers.
+
+Clients MUST download and validate the `token_metadata.sig` against the bytes of the `token_metadata.json` and the `MINTING_TRUST_ROOT` public key before attempting to process the `token_metadata.json`. Clients MUST reject the json with an error if the signature checking fails.
+
+Clients SHOULD include a minimum for `signing_timestamp` as part of their build, and reject any `token_metadata.json` which is less than the baked-in `signing_timestamp`. This prevents replay attacks where an old `token_metadata.json` document is substituted for the latest one with the goal of confusing the users by making their tokens display differently or not at all. They can fetch the latest `token_metadata.json` at build time to get the latest `signing_timestamp`.
+
+Clients MAY continue to bake token metadata into their build at their discretion, for instance, in order to support offline wallets or other cases where the client cannot access the internet. When both baked-in metadata and the `token_metadata.json` are available, the validated `token_metadata.json` should be considered a more reliable source of truth.
+
+Maintainers of the `token_metadata.json` SHOULD NOT:
+
+* Allow duplicate records for a given token id
```

Comment:
This may just be a wording issue, but I suggest we allow duplicate records for a given token id to exist across different versions of the token_metadata.json file. Requiring uniqueness within a given token_metadata.json file seems appropriate.

Reply:
This would allow for the concept of an active token_metadata.json file, and would prevent users from needing to fetch all token_metadata.json files to collect token data for all tokens.